db/seeds.rb

```diff
@@ -10,3 +10,112 @@
+                                of two Los Angeles mobsters, a gangster's wife, a boxer and two small-time criminals.",
+                  director: 'Quentin Tarantino',
+                  genre: 'crime',
+                  duration: 178 },
+                { title: 'Forrest Gump',
+                  description: 'The presidencies of Kennedy and Johnson, the Vietnam War, the Watergate scandal
+                                and other historical events unfold from the perspective of an Alabama man with
+                                an IQ of 75, whose only desire is to be reunited with his childhood sweetheart.',
+                  director: 'Robert Zemeckis',
+                  genre: 'drama',
+                  duration: 142 },
+                { title: 'Gladiator',
+                  description: 'A former Roman General sets out to exact vengeance against the corrupt
+                                emperor who murdered his family and sent him into slavery.',
+                  director: 'Ridley Scott',
+                  genre: 'historical drama',
+                  duration: 171 }
+              ])
+Rails.logger.info 'Movies created.'
+
+Rails.logger.info 'Creating Ticket Desks...'
+ticket_desk = TicketDesk.create!
+Rails.logger.info 'Ticket Desks created.'
+
+Rails.logger.info 'Creating Screenings...'
+Screening.create!([
+                    { cinema_hall: CinemaHall.all.sample, screening_time: Time.current + 1.hour,
+                      movie: Movie.all.sample },
+                    { cinema_hall: CinemaHall.all.sample, screening_time: Time.current + 2.hour,
+                      movie: Movie.all.sample },
+                    { cinema_hall: CinemaHall.all.sample, screening_time: Time.current + 3.hour,
+                      movie: Movie.all.sample },
+                    { cinema_hall: CinemaHall.all.sample, screening_time: Time.current + 4.hour,
+                      movie: Movie.all.sample }
+                  ])
+Rails.logger.info 'Screenings created.'
+
+Rails.logger.info 'Creating Reservations...'
+Reservation.create!([
+                      { status: 'pending', screening: Screening.all.sample, ticket_desk: ticket_desk, user: client },
+                      { status: 'pending', screening: Screening.all.sample, ticket_desk: ticket_desk, user: client  },
+                      { status: 'pending', screening: Screening.all.sample, ticket_desk: ticket_desk, user: client  }
+                    ])
+Rails.logger.info 'Reservations created.'
+
+Rails.logger.info 'Creating Seats Reservations...'
+letters = ('A'..'J').to_a
+Reservation.all.each do |reservation|
+  rand(1..3).times do |_number|
+    SeatsReservation.create!([row: letters[rand(1..reservation.screening.cinema_hall.row_number) - 1],
+                              seat_number: rand(1..reservation.screening.cinema_hall.row_total_seats),
+                              cinema_hall: reservation.screening.cinema_hall,
+                              reservation: reservation])
```

Comment:
Ok so, the chance for that is low, but if we would run seeds in CI then it will start happening because of the number of repetitions of that process.

You are using rand which returns a random value - as such there is really nothing preventing it to return the same row and seat_number pair twice in that 3 repetitions.

Wojciech pointed you to shuffle method, which could be used like that:

```suggestion
rows= ('A'..'J').to_a.shuffle
seats=(1..reservation.screening.cinema_hall.row_total_seats).to_a.shuffle
Reservation.all.each do |reservation|
  rand(1..3).times do |number|
    SeatsReservation.create!([row: rows[number],
                              seat_number: seats[number],
                              cinema_hall: reservation.screening.cinema_hall,
                              reservation: reservation])
```

As there are no repetitions in `seats` and `rows` array it will never return twice the same pair in that 3 repetitions.

Reply:
Implemented as discuessed:
`Reservation.all.each do |reservation|
  rows = ('A'..'J').to_a.shuffle
  seats = (1..reservation.screening.cinema_hall.row_total_seats).to_a.shuffle
  rand(1..3).times do |number|
    SeatsReservation.create!([row: rows[number],
                              seat_number: seats[number],
                              cinema_hall: reservation.screening.cinema_hall,
                              reservation: reservation])
  end
end`